examples/citi_bike/online_lp/launcher.py

```diff
@@ -0,0 +1,235 @@
+import io
+import math
+from typing import List, Tuple
+
+import argparse
+import numpy as np
+import yaml
+
+from maro.data_lib import BinaryReader, ItemTickPicker
+from maro.event_buffer import Event
+from maro.forecasting import OneStepFixWindowMA as Forecaster
+from maro.simulator import Env
+from maro.simulator.scenarios.citi_bike.adj_loader import load_adj_from_csv
+from maro.simulator.scenarios.citi_bike.common import Action, BikeReturnPayload, DecisionEvent
+from maro.simulator.scenarios.citi_bike.events import CitiBikeEvents
+from maro.utils import convert_dottable
+
+from citi_bike_lp import LP
+
+# For debug only.
+PEEP_AND_USE_REAL_DATA: bool = False
+ENV: Env = None
+TRIP_PICKER: ItemTickPicker = None
+
+class MaLpAgent():
+    def __init__(self, lp: LP, num_station: int, num_time_interval: int, ticks_per_interval: int, ma_window_size: int):
+        self._lp = lp
+        self._num_station = num_station
+        self._num_time_interval = num_time_interval
+        self._ticks_per_interval = ticks_per_interval
+        if not PEEP_AND_USE_REAL_DATA:
+            self._demand_forecaster = [Forecaster(window_size=ma_window_size) for _ in range(self._num_station)]
+            self._supply_forecaster = [Forecaster(window_size=ma_window_size) for _ in range(self._num_station)]
+            self._num_recorded_interval = 0
+            self._next_event_idx = 0
+
+    def _record_history(self, env_tick: int, finished_events: List[Event]):
+        """
+        Args:
+            env_tick (int): The current Env tick.
+            finished_events (List[Event]): The finished events got from the Env.
+        """
+        num_interval_to_record = (env_tick - 1) // self._ticks_per_interval - self._num_recorded_interval
+        if num_interval_to_record <= 0:
+            return
+        demand_history = np.zeros((num_interval_to_record, self._num_station), dtype=np.int32)
+        supply_history = np.zeros((num_interval_to_record, self._num_station), dtype=np.int32)
+
+        while self._next_event_idx < len(finished_events):
+            # Calculate the interval index of this finished event.
+            interval_idx = (
+                finished_events[self._next_event_idx].tick // self._ticks_per_interval - self._num_recorded_interval
+            )
+            assert interval_idx >= 0, "The finished events are not sorted by tick!"
+            if interval_idx >= num_interval_to_record:
+                break
+
+            # Check the event_type to get the historical demand and supply.
+            event_type = finished_events[self._next_event_idx].event_type
```

Comment:
add prefix: env_tick, env_ticks_per_decsion_interval -> const, agent_interval_idx, lp_decision_point -> relative

Reply:
Add docstrings for this agent to introduce the relationship between interval and env tick